agent/functions/put_file_test.go

```diff
@@ -0,0 +1,193 @@
+package functions
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github/clover0/github-issue-agent/store"
+)
+
+func TestPutFile(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       PutFileInput
+		wantErr     bool
+		checkFile   bool
+		setupFunc   func(t *testing.T, dir string) // テスト前のセットアップ
+		validateFunc func(t *testing.T, got store.File, err error)
+	}{
+		{
+			name: "正常系：新規ファイル作成",
+			input: PutFileInput{
+				OutputPath:  "test.txt",
+				ContentText: "test content",
+			},
+			checkFile: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				if got.Path != "test.txt" || got.Content != "test content\n" {
+					t.Errorf("unexpected result: got %+v", got)
+				}
+			},
+		},
+		{
+			name: "正常系：ディレクトリ作成を伴うファイル作成",
+			input: PutFileInput{
+				OutputPath:  "new/dir/test.txt",
+				ContentText: "nested content",
+			},
+			checkFile: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				if got.Path != "new/dir/test.txt" || got.Content != "nested content\n" {
+					t.Errorf("unexpected result: got %+v", got)
+				}
+			},
+		},
+		{
+			name: "正常系：既存ファイルの上書き",
+			input: PutFileInput{
+				OutputPath:  "existing.txt",
+				ContentText: "new content",
+			},
+			setupFunc: func(t *testing.T, dir string) {
+				path := filepath.Join(dir, "existing.txt")
+				if err := os.WriteFile(path, []byte("old content"), 0644); err != nil {
+					t.Fatal(err)
+				}
+			},
+			checkFile: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				if got.Path != "existing.txt" || got.Content != "new content\n" {
+					t.Errorf("unexpected result: got %+v", got)
+				}
+			},
+		},
+		{
+			name: "異常系：無効なパス（ファイル名なし）",
+			input: PutFileInput{
+				OutputPath:  "/invalid/path/",
+				ContentText: "test content",
+			},
+			wantErr: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err == nil {
+					t.Error("expected error but got nil")
+				}
+			},
+		},
+		{
+			name: "異常系：書き込み権限なしディレクトリ",
+			input: PutFileInput{
+				OutputPath:  "noperm/test.txt",
+				ContentText: "test content",
+			},
+			setupFunc: func(t *testing.T, dir string) {
+				noPermDir := filepath.Join(dir, "noperm")
+				if err := os.Mkdir(noPermDir, 0000); err != nil {
+					t.Fatal(err)
+				}
+			},
+			wantErr: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err == nil {
+					t.Error("expected error but got nil")
```

Comment:
from security_reviewer
パーミッションテストは良い取り組みですが、シンボリックリンクを使用した攻撃に対するテストケースが不足しています。悪意のあるシンボリックリンクを介したファイル操作のテストを追加すべきです。

```suggestion
以下のようなテストケースを追加することを推奨します：
```go
{
    name: "異常系：シンボリックリンク攻撃",
    input: PutFileInput{
        OutputPath: "symlink_target.txt",
        ContentText: "test content",
    },
    setupFunc: func(t *testing.T, dir string) {
        // 重要なファイルへのシンボリックリンクを作成
        os.Symlink("/etc/passwd", "symlink_target.txt")
    },
    wantErr: true,
},
```
```